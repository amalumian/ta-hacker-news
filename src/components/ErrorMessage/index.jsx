import { Alert } from 'antd'
import PropTypes from 'prop-types'

import './index.css'

const ErrorMessage = ({ message }) => {
  return <Alert className='error-message' message={message} type='error' />
}

ErrorMessage.propTypes = {
  message: PropTypes.string.isRequired,
}

export default ErrorMessage
