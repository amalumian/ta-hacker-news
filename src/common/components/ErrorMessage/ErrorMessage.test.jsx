import { describe, expect, test } from 'vitest'
import { render, screen, within } from '@testing-library/react'

import ErrorMessage from './index'

describe('ErrorMessage Component', () => {
  test('renders with correct message', () => {
    const errorMessage = 'Error loading story'
    render(<ErrorMessage message={errorMessage} />)

    const alertElement = screen.getByRole('alert')
    const alertMessage = within(alertElement).getByText(errorMessage)

    expect(alertMessage).toBeInTheDocument()
  })

  test('matches snapshot', () => {
    const { asFragment } = render(<ErrorMessage message='Snapshot test' />)
    expect(asFragment()).toMatchSnapshot()
  })
})
