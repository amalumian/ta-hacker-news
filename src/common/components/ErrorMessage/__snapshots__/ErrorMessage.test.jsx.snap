// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ErrorMessage Component > matches snapshot 1`] = `
<div
  class="ant-alert ant-alert-error ant-alert-no-icon error-message css-dev-only-do-not-override-d2lrxs"
  data-show="true"
  role="alert"
>
  <div
    class="ant-alert-content"
  >
    <div
      class="ant-alert-message"
    >
      Snapshot test
    </div>
  </div>
</div>
`;
